tests/src/Mod/Part/App/FeaturePartCommon.cpp

```diff
@@ -0,0 +1,209 @@
+
+    // Assert
+    // If we wanted to be excessive, we could check the bounds:
+    // EXPECT_EQ(bb.MinX, 0);
+    // EXPECT_EQ(bb.MinY, 1);
+    // EXPECT_EQ(bb.MinZ, 0);
+    // EXPECT_EQ(bb.MaxX, 1);
+    // EXPECT_EQ(bb.MaxY, 2);
+    // EXPECT_EQ(bb.MaxZ, 3);
+    double volume = PartTestHelpers::getVolume(ts.getShape());
+    EXPECT_DOUBLE_EQ(volume, 3.0);
+}
+
+TEST_F(FeaturePartCommonTest, testNonIntersecting)
+{
+    // Arrange
+    _common->Base.setValue(_box1obj);
+    _common->Tool.setValue(_box3obj);
+
+    // Act
+    _common->execute();
+    Part::TopoShape ts = _common->Shape.getValue();
+    Base::BoundBox3d bb = ts.getBoundBox();
+
+    // Assert
+    EXPECT_FALSE(bb.IsValid());
+    double volume = PartTestHelpers::getVolume(ts.getShape());
+    EXPECT_DOUBLE_EQ(volume, 0.0);
+}
+
+TEST_F(FeaturePartCommonTest, testTouching)
+{
+    // Arrange
+    _common->Base.setValue(_box1obj);
+    _common->Tool.setValue(_box4obj);
+
+    // Act
+    _common->execute();
+    Part::TopoShape ts = _common->Shape.getValue();
+    Base::BoundBox3d bb = ts.getBoundBox();
+
+    // Assert
+    EXPECT_FALSE(bb.IsValid());
+    double volume = PartTestHelpers::getVolume(ts.getShape());
+    EXPECT_DOUBLE_EQ(volume, 0.0);
+}
+
+TEST_F(FeaturePartCommonTest, testAlmostTouching)
+{
+    // Arrange
+    _common->Base.setValue(_box1obj);
+    _common->Tool.setValue(_box5obj);
+
+    // Act
+    _common->execute();
+    Part::TopoShape ts = _common->Shape.getValue();
+    Base::BoundBox3d bb = ts.getBoundBox();
+
+    // Assert
+    EXPECT_FALSE(bb.IsValid());
+    double volume = PartTestHelpers::getVolume(ts.getShape());
+    EXPECT_DOUBLE_EQ(volume, 0.0);
+}
+
+TEST_F(FeaturePartCommonTest, testBarelyIntersecting)
+{
+    // Arrange
+    _common->Base.setValue(_box1obj);
+    _common->Tool.setValue(_box6obj);
+
+    // Act
+    _common->execute();
+    Part::TopoShape ts = _common->Shape.getValue();
+    Base::BoundBox3d bb = ts.getBoundBox();
+
+    // Assert
+    EXPECT_EQ(bb.MinX, 0);
+    EXPECT_EQ(bb.MinY, 1.9999);
+    EXPECT_EQ(bb.MinZ, 0);
+    EXPECT_EQ(bb.MaxX, 1);
+    EXPECT_EQ(bb.MaxY, 2);
+    EXPECT_EQ(bb.MaxZ, 3);
+    double volume = PartTestHelpers::getVolume(ts.getShape());
+    double target = Base::Precision::Confusion() * 3 * 1000;
+    // FLOAT, not DOUBLE here, because ULP accuracy would be too precise.
+    EXPECT_FLOAT_EQ(volume, target);  // 0.00029999999999996696);
+}
+
+TEST_F(FeaturePartCommonTest, testMustExecute)
+{
+    // Act
+    short mE = _common->mustExecute();
+    // Assert
+    EXPECT_FALSE(mE);
+    _common->Base.setValue(_box1obj);
+    // Assert
+    mE = _common->mustExecute();
+    EXPECT_FALSE(mE);
+    // Act
+    _common->Tool.setValue(_box2obj);
+    // Assert
+    mE = _common->mustExecute();
+    EXPECT_TRUE(mE);
+    _doc->recompute();
+    mE = _common->mustExecute();
+    EXPECT_FALSE(mE);
+}
+
+TEST_F(FeaturePartCommonTest, testGetProviderName)
+{
+    // Act
+    _common->execute();
```

Comment:
I think https://google.github.io/googletest/primer.html#invoking-the-tests says that the state is always reset.

Correct that _common is not fully initialized here; however this test should and does pass even with a partially initialized object.  I'm trying to match wart-for-wart with classical FC code so I'm doing it this way since I think we'd want to know if it changed.  At that point, someone can argue for changing the test if it is appropriate.  Open to opinions here.

Reply:
Great, leave it as-is then